addons/mail/static/src/models/channel_member.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear, replace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'ChannelMember',
+    identifyingFields: ['id'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeChannelAsOfflineMember() {
+            return this.partner && !this.partner.isOnline ? replace(this.channel) : clear();
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeChannelAsOnlineMember() {
+            return this.partner && this.partner.isOnline ? replace(this.channel) : clear();
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeName() {
+            return this.partner.nameOrDisplayName;
+        },
+    },
+    fields: {
+        channel: one('Channel', {
+            inverse: 'channelMembers',
+            readonly: true,
+            required: true,
+        }),
+        channelAsOfflineMember: one('Channel', {
+            compute: '_computeChannelAsOfflineMember',
+            inverse: 'orderedOfflineMembers',
+        }),
+        channelAsOnlineMember: one('Channel', {
+            compute: '_computeChannelAsOnlineMember',
+            inverse: 'orderedOnlineMembers',
+        }),
+        channelMemberView: one('ChannelMemberView', {
```

Comment:
This side of the relation should be a many, there can always be multiple views for a record